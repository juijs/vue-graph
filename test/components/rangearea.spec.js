import { mount } from '@vue/test-utils'
import GraphRangeArea from '@/components/rangearea'

describe('rangearea.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphRangeArea, {
            props: {
                width: 600,
                height: 400,
                axisFullMode: true,
                labels: [ "2001", "2002", "2003", "2004", "2005", "2006", "2007", "2008" ],
                values: [
                    [ 4, 6, 8 ],
                    [ 5, 7, 9 ],
                    [ 6, 8, 11 ],
                    [ 5, 6, 10 ],
                    [ 2, 3, 6 ],
                    [ 5, 6, 8 ],
                    [ 2, 4, 6 ],
                    [ 8, 10, 12 ]
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});