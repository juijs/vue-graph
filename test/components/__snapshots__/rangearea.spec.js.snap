// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rangearea.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="400"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="600"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="300"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="500"
        x="0"
        y="0"
      />
      <g
        class="grid-fullblock"
        transform="translate(50,350)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2001
          </text>
        </g>
        <g
          transform="translate(71.42857142857143,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2002
          </text>
        </g>
        <g
          transform="translate(142.85714285714286,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2003
          </text>
        </g>
        <g
          transform="translate(214.28571428571428,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2004
          </text>
        </g>
        <g
          transform="translate(285.7142857142857,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2005
          </text>
        </g>
        <g
          transform="translate(357.14285714285717,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2006
          </text>
        </g>
        <g
          transform="translate(428.5714285714286,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2007
          </text>
        </g>
        <g
          transform="translate(500.00000000000006,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2008
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="500"
          y1="0"
          y2="0"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-range"
        transform="translate(50,50)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            12
          </text>
        </g>
        <g
          transform="translate(0,50)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            10
          </text>
        </g>
        <g
          transform="translate(0,100)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            8
          </text>
        </g>
        <g
          transform="translate(0,150)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            6
          </text>
        </g>
        <g
          transform="translate(0,200)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            4
          </text>
        </g>
        <g
          transform="translate(0,250)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            2
          </text>
        </g>
        <g
          transform="translate(0,300)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            0
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="0"
          y1="0"
          y2="300"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="300"
          stroke="transparent"
          width="500"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-rangearea"
        transform="translate(50,50)"
      >
        <polygon
          fill="#7977C2"
          fill-opacity="0.5"
          points="0,200 71.42857142857143,175 142.85714285714286,150 214.28571428571428,175 285.7142857142857,250 357.14285714285717,175 428.5714285714286,250 500.00000000000006,100 500.00000000000006,0 428.5714285714286,150 357.14285714285717,100 285.7142857142857,150 214.28571428571428,50 142.85714285714286,25 71.42857142857143,75 0,100 0,200"
          stroke-width="0"
        />
      </g>
      <g
        class="brush-line"
        transform="translate(50,50)"
      >
        <g>
          <path
            cursor="normal"
            d="M0,150 L71.42857142857143,125 L142.85714285714286,100 L214.28571428571428,150 L285.7142857142857,225 L357.14285714285717,150 L428.5714285714286,200 L500.00000000000006,50"
            fill="transparent"
            stroke="#7977C2"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="2"
            style="pointer-events:stroke"
            x1="0"
            x2="500.00000000000006"
          />
        </g>
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
